test/unit/exec/src/IndependentElements.cpp

```diff
@@ -0,0 +1,204 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include <iostream>
+#include <random>
+#include <type_traits>
+
+#if BOOST_COMP_MSVC
+// MSVC uses __restrict instead of __restrict__.
+#    define __restrict__ __restrict
+#endif
+
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wexit-time-destructors"
+#endif
+// Global Host object used by all tests.
+using Host = alpaka::DevCpu;
+static Host host = alpaka::getDevByIdx(alpaka::PlatformCpu{}, 0);
+#if BOOST_COMP_CLANG
+#    pragma clang diagnostic pop
+#endif
+
+/* Add the group id to te value of each element in the group.
+ * Each group is composed by the elements first[group]..first[group+1]-1 .
+ */
+struct IndependentWorkKernel
+{
+    template<typename TAcc, typename T>
+    ALPAKA_FN_ACC void operator()(
+        TAcc const& acc,
+        T const* __restrict__ in,
+        T* __restrict__ out,
+        alpaka::Idx<TAcc> const* __restrict__ indices,
+        alpaka::Idx<TAcc> groups) const
+    {
+        using Idx = alpaka::Idx<TAcc>;
+
+        for(auto group : alpaka::independentGroups(acc, groups))
+        {
+            Idx first = indices[group];
+            Idx last = indices[group + 1];
+            Idx size = last - first;
+            for(auto index : alpaka::independentGroupElements(acc, size))
+            {
+                out[first + index] = in[first + index] + static_cast<float>(group);
+            }
+        }
+    }
+};
+
+/* Test the IndependentWorkKernel kernel on all devices
+ */
+template<typename TAcc, typename TKernel>
+void testIndependentWorkKernel(
+    alpaka::Idx<TAcc> groups,
+    alpaka::Idx<TAcc> grid_size,
+    alpaka::Idx<TAcc> block_size,
+    TKernel kernel)
+{
+    using Acc = TAcc;
+    using Idx = alpaka::Idx<Acc>;
+    using Platform = alpaka::Platform<Acc>;
+    using Device = alpaka::Dev<Platform>;
+    using Queue = alpaka::Queue<Device, alpaka::NonBlocking>;
+
+    // Initialise the accelerator platform.
+    Platform platform{};
+
+    // Random number generator with a gaussian distribution.
+    std::random_device rd{};
+    std::default_random_engine engine{rd()};
+
+    // Uniform distribution.
+    std::uniform_int_distribution<Idx> random_size{100, 201};
+
+    // Gaussian distribution.
+    std::normal_distribution<float> dist{0., 1.};
+
+    // Build the groups.
+    auto indices_h = alpaka::allocMappedBuf<Idx, Idx>(host, platform, groups + 1);
+    indices_h[0] = 0;
+    for(Idx i = 0; i < groups; ++i)
+    {
+        // Group "i" has "size" elements.
+        auto size = random_size(engine);
+        indices_h[i + 1] = indices_h[i] + size;
+    }
+
+    // Tolerance.
+    constexpr float epsilon = 0.000001f;
+
+    // Buffer size.
+    const Idx size = indices_h[groups];
+
+    // Allocate the input and output host buffer in pinned memory accessible by the Platform devices.
+    auto in_h = alpaka::allocMappedBuf<float, Idx>(host, platform, size);
```

Comment:
Camel Case variable naming?
```suggestion
    auto inH = alpaka::allocMappedBuf<float, Idx>(host, platform, size);
```

Reply:
I don't particularly like this naming style.
I follow it for user-facing function names, like in the rest of alpaka, but I don't like to use it for variable names.

Reply:
Yes, i also don't like actually.